src/encoding/text.rs

```diff
@@ -742,6 +746,24 @@ mod tests {
         parse_with_python_client(encoded);
     }
 
+    #[test]
+    fn encode_histogram_family_with_empty_family_labels() {
+        let mut registry = Registry::default();
+        let family =
+            Family::new_with_constructor(|| Histogram::new(exponential_buckets(1.0, 2.0, 10)));
+        registry.register("my_histogram", "My histogram", family.clone());
+
+        family
+            .get_or_create(&())
```

Comment:
In what use-case would you want a `Family` with a `()` as a label set?

To remove this footgun, shall we remove the `EncodeLabelSet` implementation for `()`?

https://github.com/prometheus/client_rust/blob/23c31bdacdc2445b0b85e79684a6d2534330fc29/src/encoding.rs#L352-L356

Reply:
Now that I understand that the `Family` wrapper is optional, I agree that using `()` as the label set doesn't make much sense.

However, when I originally ran into this issue I was using an empty struct, which has similar behavior. I was intending to go back and add any needed labels to the struct later but noticed this issue before I could do that. I switched to `()` when I was simplifying my code for posting here.

Removing the `EncodeLabelSet` implementation from `()` seems like a good start, but ideally a fix would cover the empty struct case as well.

Reply:
I took a look at removing the `EncodeLabelSet` implementation for `()`, but it looks like that impl is relied on in at least these four places:

https://github.com/prometheus/client_rust/blob/1f3923a7a53db30081a11d1741a1c6e50ae08111/src/encoding/text.rs#L189

https://github.com/prometheus/client_rust/blob/1f3923a7a53db30081a11d1741a1c6e50ae08111/src/encoding/text.rs#L213

https://github.com/prometheus/client_rust/blob/1f3923a7a53db30081a11d1741a1c6e50ae08111/src/encoding/text.rs#L269

https://github.com/prometheus/client_rust/blob/1f3923a7a53db30081a11d1741a1c6e50ae08111/src/encoding/text.rs#L276

Reply:
I would assume that we can replace `()` with a private uninhabited type that implements `EncodeLabelSet`. Something along the lines of:

``` rust
enum Never {}

impl EncodeLabelSet for Never {
  // ...
}

// ...

// Then in the above cases:
self.encode_labels::<Never>(None)?;
```

Want to give this a shot @tylerlevine ?

Reply:
Good idea. Yes, let me give it a try.

Reply:
Friendly ping @tylerlevine. What do you think of the suggestion above?

Reply:
@mxinden Looks like that works fine. I cleaned up a bit and updated the PR.